lib/ui/window/platform_configuration_unittests.cc

```diff
@@ -15,10 +15,166 @@
+              (const, override));
+  MOCK_METHOD(void, SendChannelUpdate, (std::string, bool), (override));
+  MOCK_METHOD(double,
+              GetScaledFontSize,
+              (double font_size, int configuration_id),
+              (const, override));
+};
+
+class MockPlatformMessageHandler : public PlatformMessageHandler {
+ public:
+  MOCK_METHOD(void,
+              HandlePlatformMessage,
+              (std::unique_ptr<PlatformMessage> message),
+              (override));
+  MOCK_METHOD(bool,
+              DoesHandlePlatformMessageOnPlatformThread,
+              (),
+              (const, override));
+  MOCK_METHOD(void,
+              InvokePlatformMessageResponseCallback,
+              (int response_id, std::unique_ptr<fml::Mapping> mapping),
+              (override));
+  MOCK_METHOD(void,
+              InvokePlatformMessageEmptyResponseCallback,
+              (int response_id),
+              (override));
+};
+
+// A class that can launch a RuntimeController with the specified
+// RuntimeDelegate.
+//
+// To use this class, contruct this class with Create, call LaunchRootIsolate,
+// and get the runtime controller with Controller().
+class RuntimeControllerContext {
+ public:
+  using VoidCallback = std::function<void()>;
+
+  [[nodiscard]] static std::unique_ptr<RuntimeControllerContext> Create(
+      Settings settings,                //
+      const TaskRunners& task_runners,  //
+      RuntimeDelegate& client) {
+    auto [vm, isolate_snapshot] = Shell::InferVmInitDataFromSettings(settings);
+    FML_CHECK(vm) << "Must be able to initialize the VM.";
+    // Construct the class with `new` because `make_unique` has no access to the
+    // private constructor.
+    RuntimeControllerContext* raw_pointer = new RuntimeControllerContext(
+        settings, task_runners, client, std::move(vm), isolate_snapshot);
+    return std::unique_ptr<RuntimeControllerContext>(raw_pointer);
+  }
+
+  ~RuntimeControllerContext() {
+    PostSync(task_runners_.GetUITaskRunner(),
+             [&]() { runtime_controller_.reset(); });
+  }
+
+  // Launch the root isolate. The post_launch callback will be executed in the
+  // same UI task, which can be used to create initial views.
+  void LaunchRootIsolate(RunConfiguration& configuration,
+                         VoidCallback post_launch) {
+    PostSync(task_runners_.GetUITaskRunner(), [&]() {
+      bool launch_success = runtime_controller_->LaunchRootIsolate(
+          settings_,                                  //
+          []() {},                                    //
+          configuration.GetEntrypoint(),              //
+          configuration.GetEntrypointLibrary(),       //
+          configuration.GetEntrypointArgs(),          //
+          configuration.TakeIsolateConfiguration());  //
+      ASSERT_TRUE(launch_success);
+      post_launch();
+    });
+  }
+
+  // Get the runtime controller.
+  RuntimeController& Controller() { return *runtime_controller_; }
```

Comment:
```suggestion
  RuntimeController& controller() { return *runtime_controller_; }
```

Reply:
I think this rule is rarely applied in the engine repo and is redeemed weird. We've been using the CamelCase for all methods, such as in https://github.com/flutter/engine/blob/main/shell/common/engine.h#L952C28-L952C37

Reply:
Hm it looks like the engine is inconsistent here, but it looks like `PlatformConfiguration` uses variable naming for accessors. For example:

https://github.com/flutter/engine/blob/494fd7fe85a30ce4dfc28058961bb71920f514cc/lib/ui/window/platform_configuration.h#L283

I would be stay consistent with `PlatformConfiguration`'s style given this is what this code tests.

Reply:
But `PlatformConfiguration` also has `GetAssetManager` `GetFontCollection` `GetPersistentIsolateData`...

Reply:
I've refactored the code to some methods with callbacks that provide the controller. This is mostly to ensure that the tasks are performed on the UI thread, but also voids this concern.